**Q-9.** Given the score of students in multiple exams

    Name    Exam 1    Exam 2    Final Exam
--------------------------------------

    Karan     85        90         92
    Deepa     70        80         85
    Karthik   90        85         88
    Chandan   75        70         75
    Jeevan    90        92         96

Test the hypothesis that the mean scores of all the students are the same. If not, name the
student with the highest score.

In [1]:
import scipy.stats as stats
import numpy as np
Karan = [85, 90, 92]
Deepa = [70, 80, 85]
Karthik = [90, 85, 88]
Chandan = [75, 70, 75]
Jeevan = [90, 92, 96]
lst=[Karan,Deepa,Karthik,Chandan,Jeevan]
# Perform ANOVA test
f_value, p_value = stats.f_oneway(Karan,Deepa,Karthik,Chandan,Jeevan)

# Print results
print("F-value:", f_value)
print("p-value:", p_value)
for i in lst:
    print(f'{i}-------->{np.mean(i)}') 

F-value: 10.197552447552443
p-value: 0.0014839499389315998
[85, 90, 92]-------->89.0
[70, 80, 85]-------->78.33333333333333
[90, 85, 88]-------->87.66666666666667
[75, 70, 75]-------->73.33333333333333
[90, 92, 96]-------->92.66666666666667


The ANOVA test results indicate that the calculated F-value is approximately 10.11 and the p-value is approximately 0.001. Since the p-value is less than the commonly chosen significance level of 0.05, we reject the null hypothesis. This means that we have enough evidence to conclude that there is a significant difference in means among the groups.Jeevan is having highest score(92.67).